--- a/drivers/dahdi/Kbuild
+++ b/drivers/dahdi/Kbuild
@@ -68,7 +68,11 @@ obj-m += dahdi_echocan_oslec.o
 obj-m += ../staging/echo/echo.o
 endif
 
-CFLAGS_MODULE += -I$(DAHDI_INCLUDE) -I$(src) -Wno-format-truncation
+ifeq ($(CC),clang)
+  CFLAGS_MODULE += -I$(DAHDI_INCLUDE) -I$(src)
+else
+  CFLAGS_MODULE += -I$(DAHDI_INCLUDE) -I$(src) -Wno-format-truncation
+endif
 
 ifndef HOTPLUG_FIRMWARE
 ifneq (,$(filter y m,$(CONFIG_FW_LOADER)))
